mitodl/mitxonline
# Adds flexible pricing flag to Discount objects

#### Pre-Flight checklist


#### What are the relevant tickets?

#674 

#### What's this PR do?

Adds a "for flexible pricing" flag to the Discount model to flag codes that should only be used for flexible pricing. Discounts with this flag set won't be redeemable through the front end.

#### How should this be manually tested?

All automated tests should pass.

1. Add (or modify) a discount code and set the "For flexible pricing" flag. The flag should be set by default.
2. Add something to your cart.
3. Attempt to apply the discount code from step 1. You should get the standard "code is invalid" message.
4. Assign the discount to a flexible pricing tier.
5. Go through the Flexible Pricing request process such that you would end up in the tier you used in step #4. 
6. Attempt to purchase the course again. The flexible pricing discount should apply.
